artichoke-backend/src/extn/core/string/mod.rs

```diff
@@ -59,12 +59,14 @@ impl BoxUnboxVmValue for String {
         let encoding = (encoding_flag >> ENCODING_FLAG_BITPOS) as u8;
         let encoding = Encoding::try_from_flag(encoding).map_err(|_| TypeError::with_message("Unknown encoding"))?;
 
-        let mut s = String::from_raw_parts(RawParts {
-            ptr: ptr.cast::<u8>(),
-            length,
-            capacity,
-        });
-        s.set_encoding(encoding);
+        let s = String::from_raw_parts_with_encoding(
+            RawParts {
+                ptr: ptr.cast::<u8>(),
+                length,
+                capacity,
+            },
+            encoding,
+        );
```

Comment:
@lopopolo Come to think of it, does it make sense to implement `force_encoding` purely so there isn't the breaking change of removing `set_encoding`? I had originally removed `set_encoding` since I didn't think it'd be required anymore, but since `String#force_encoding` is a thing, maybe it's still needed?